migration-engine/connectors/sql-migration-connector/src/sql_database_migration_inferrer.rs

```diff
@@ -81,11 +97,57 @@ fn infer(
     current_database_schema: SqlSchema,
     expected_database_schema: SqlSchema,
     flavour: &dyn SqlFlavour,
+    next_datamodel: &Datamodel,
 ) -> SqlMigration {
     let steps =
         sql_schema_differ::calculate_steps(Pair::new(&current_database_schema, &expected_database_schema), flavour);
+    let next_schema = &expected_database_schema;
+
+    let added_columns_with_virtual_defaults: Vec<(usize, usize)> = steps
+        .iter()
+        .filter_map(|step| step.as_alter_table())
+        .flat_map(move |alter_table| {
+            alter_table
+                .changes
+                .iter()
+                .filter_map(|change| change.as_add_column())
+                .map(move |column| -> (usize, usize) { (*alter_table.table_index.next(), column.column_index) })
```

Comment:
It's quite obviously `(usize, usize)`, do we need to annotate the type info here?

Reply:
I did that to avoid the confusion between `(usize, usize)`, `(&usize, usize)`, etc. It's quite easy to introduce a type mismatch and the I'm scared of what the compiler error would be.

Reply:
Oh, yeah iterators and `Copy` items are weird. I guess in rust type system an `Iterator` cannot just `Copy` even if the items are `Copy`...